From 8cf270451b9a14fc73a6ca32d81febfeae026c30 Mon Sep 17 00:00:00 2001
Message-Id: <8cf270451b9a14fc73a6ca32d81febfeae026c30.1559359751.git.jan.steffens@gmail.com>
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Sat, 1 Jun 2019 05:27:43 +0200
Subject: [PATCH] libgail-util: Fix symbol visibility with meson

Autotools didn't add -fvisibility=hidden to the build of libgail-util-3.
Meson does and produced a library that exported no symbols.
---
 libgail-util/gailmisc.h     | 7 +++++++
 libgail-util/gailtextutil.h | 6 ++++++
 2 files changed, 13 insertions(+)

diff --git a/libgail-util/gailmisc.h b/libgail-util/gailmisc.h
index a0145ab180..b6a160128d 100644
--- a/libgail-util/gailmisc.h
+++ b/libgail-util/gailmisc.h
@@ -24,47 +24,54 @@
 
 G_BEGIN_DECLS
 
+GDK_AVAILABLE_IN_ALL
 AtkAttributeSet* gail_misc_add_attribute          (AtkAttributeSet   *attrib_set,
                                                    AtkTextAttribute   attr,
                                                    gchar             *value);
+GDK_AVAILABLE_IN_ALL
 AtkAttributeSet* gail_misc_layout_get_run_attributes
                                                   (AtkAttributeSet   *attrib_set,
                                                    PangoLayout       *layout,
                                                    const gchar       *text,
                                                    gint              offset,
                                                    gint              *start_offset,
                                                    gint              *end_offset);
 
+GDK_AVAILABLE_IN_ALL
 AtkAttributeSet* gail_misc_get_default_attributes (AtkAttributeSet   *attrib_set,
                                                    PangoLayout       *layout,
                                                    GtkWidget         *widget);
 
+GDK_AVAILABLE_IN_ALL
 void             gail_misc_get_extents_from_pango_rectangle
                                                   (GtkWidget         *widget,
                                                    PangoRectangle    *char_rect,
                                                    gint              x_layout,
                                                    gint              y_layout,
                                                    gint              *x,
                     		                   gint              *y,
                                                    gint              *width,
                                                    gint              *height,
                                                    AtkCoordType      coords);
 
+GDK_AVAILABLE_IN_ALL
 gint             gail_misc_get_index_at_point_in_layout
                                                   (GtkWidget         *widget,
                                                    PangoLayout       *layout, 
                                                    gint              x_layout,
                                                    gint              y_layout,
                                                    gint              x,
                                                    gint              y,
                                                    AtkCoordType      coords);
 
+GDK_AVAILABLE_IN_ALL
 void		 gail_misc_get_origins            (GtkWidget         *widget,
                                                    gint              *x_window,
 					           gint              *y_window,
 					           gint              *x_toplevel,
 					           gint              *y_toplevel);
 
+GDK_AVAILABLE_IN_ALL
 AtkAttributeSet* gail_misc_buffer_get_run_attributes
                                                   (GtkTextBuffer     *buffer,
                                                    gint              offset,
diff --git a/libgail-util/gailtextutil.h b/libgail-util/gailtextutil.h
index 7d8cf9ec20..5cc1a537fb 100644
--- a/libgail-util/gailtextutil.h
+++ b/libgail-util/gailtextutil.h
@@ -62,20 +62,26 @@ struct _GailTextUtilClass
   GObjectClass parent_class;
 };
 
+GDK_AVAILABLE_IN_ALL
 GType         gail_text_util_get_type      (void);
+GDK_AVAILABLE_IN_ALL
 GailTextUtil* gail_text_util_new           (void);
 
+GDK_AVAILABLE_IN_ALL
 void          gail_text_util_text_setup    (GailTextUtil    *textutil,
                                             const gchar     *text);
+GDK_AVAILABLE_IN_ALL
 void          gail_text_util_buffer_setup  (GailTextUtil    *textutil,
                                             GtkTextBuffer   *buffer);
+GDK_AVAILABLE_IN_ALL
 gchar*        gail_text_util_get_text      (GailTextUtil    *textutil,
                                              gpointer        layout,
                                             GailOffsetType  function,
                                             AtkTextBoundary boundary_type,
                                             gint            offset,
                                             gint            *start_offset,
                                             gint            *end_offset);
+GDK_AVAILABLE_IN_ALL
 gchar*        gail_text_util_get_substring (GailTextUtil    *textutil,
                                             gint            start_pos,
                                             gint            end_pos);
-- 
2.21.0

